core/peerset/src/lib.rs

```diff
@@ -103,201 +172,176 @@ pub struct PeersetConfig {
 ///
 /// Implements the `Stream` trait and can be polled for messages. The `Stream` never ends and never
 /// errors.
-pub struct PeersetMut {
-	parent: Arc<Peerset>,
-	rx: mpsc::UnboundedReceiver<Message>,
+#[derive(Debug)]
+pub struct Peerset {
+	data: PeersetData,
+	handle: PeersetHandle,
+	rx: mpsc::UnboundedReceiver<Action>,
+	message_queue: VecDeque<Message>,
+}
+
+impl ops::Deref for Peerset {
+	type Target = PeersetHandle;
+
+	fn deref(&self) -> &Self::Target {
+		&self.handle
+	}
 }
 
 impl Peerset {
 	/// Builds a new peerset from the given configuration.
-	pub fn from_config(config: PeersetConfig) -> (Arc<Peerset>, PeersetMut) {
+	pub fn from_config(config: PeersetConfig) -> Peerset {
```

Comment:
Being able to create handles from the `Peerset` means that we will forever have to keep in the `Peerset` the building blocks for a `PeersetHandler` unless we break the API. Whereas if we return a tuple, they can be separate.

Reply:
you can always create `PeersetHandle` by cloning an existing one

Reply:
I'm not strongly opinionated here. For me, this method could also return a tuple. But I see nothing bad, in `Peerset` keeping the building blocks for the `PeersetHandler`